app/AppKernel.php

```diff
@@ -72,7 +72,7 @@ public function __construct($environment, $debug)
          */
         try {
             list($version, $sha) = explode('@', \PackageVersions\Versions::getVersion('shopware/shopware'));
-            $this->release['version'] = $version;
+            $this->release['version'] = preg_replace('/[^0-9.]/', '', $version);
```

Comment:
I suppose that's current solution is OK since this is a dot within character class.

I've rather re-checked and should be fine:
https://regexr.com/3tape
https://stackoverflow.com/a/19976308/4827632

There should be written tests for that but unfortunately this package removes `/tests` directory and links that big thing from Shopware repository :disappointed: 

---

Ad. code extraction - I agree, but I'd extract maybe just into AppKernel's private method. Or what place would you suggest?

Thanks

Reply:
Thanks for your double check, looks like I learned something today! 💙 

Yes, I thought that adding real tests could be a problem. I am open for suggestions on how to fix this (first thought: some directory that get's itself linked into the `tests` directory, after that got linked), but I would be fine with some `protected`/`private` function.

Reply:
You're welcome, glad I could help :-)
Here is the refactoring: 0924ffc 

Ad. tests - I will be thinking and if find something, will make a merge request :peace_symbol: